docs/docs/how-to/plugins-and-themes/adding-gatsby-image-support.md

```diff
@@ -0,0 +1,261 @@
+
+The arguments:
+
+- `resolverFunction`: the resolver function that you created at step 1. It receives the node and the arguments and should return the image data object.
+- `additionalArgs`: an object defining additional args, in the same format used by [Gatsby Type Builders](https://www.gatsbyjs.com/docs/reference/graphql-data-layer/schema-customization/#gatsby-type-builders)
+
+For example, to add a `gatsbyImageData` resolver onto a `ProductImage` node that you have previously defined:
+
+```js:title=gatsby-source-example/gatsby-node.js
+// Note the different import
+import { getGatsbyImageResolver } from "gatsby-plugin-image/graphql-utils"
+
+export function createResolvers({ createResolvers }) {
+  createResolvers({
+    ProductImage: {
+      // loadImageData is your custom resolver, defined at step 2
+      gatsbyImageData: getGatsbyImageResolver(loadImageData, {
+        quality: "Int",
+      }),
+    },
+  })
+}
+```
+
+## Adding a custom image component
+
+If you have a URL-based image API, you can create a custom image component that wraps `<GatsbyImage />` and displays images generated at runtime. If you have a source plugin, this could accept your native image object, but it could equally just take a base URL and generate the image based on that. This is a good solution for image CDNs that aren't handling their own CMS data, and can generate a transformed image from just a source URL and dimensions.
+
+There are three steps to create a custom image component:
+
+1. [Create your URL builder function](#create-your-url-builder-function)
+2. [Create your image data function](#create-your-image-data-function)
+3. [Create your wrapper component](#create-your-wrapper-component) (optional)
+
+### Create your URL builder function
+
+This is similar to the `generateImageSource` described above, but just returns a URL string. This is an example for the same image host:
+
+```js
+function urlBuilder({ baseUrl, width, height, format, options }) {
+  return `https://myexampleimagehost.com/${baseURL}?w=${width}&h=${height}&fmt=${format}&q=${options.quality}`
+}
+```
+
+If your host supports it, we recommend using auto-format to deliver next-generation image formats to supported browsers. In this case, ignore the `format` option.
+
+### Create your image data function
+
+This is a similar to the image resolver described above. However because it executes in the browser, the functions that you use should be fast, and synchronous, and you can't use node APIs. You will not be downloading and generating base64 placeholders, for example, or calculating dominant colors. If you have these values pre-calculated then you can pass these in and use them, but they need to be available in the props that you pass to the function at runtime.
```

Comment:
```suggestion
This is similar to the image resolver described above. However, because it executes in the browser, the functions that you use should be fast and synchronous. That means you can't use node APIs. You will not be downloading and generating base64 placeholders, for example, or calculating dominant colors. If you have these values pre-calculated then you can pass them in, but they need to be available in the props that you pass to the function at runtime.
```

Reply:
Not quite. The reason you can't use Node APIs is that you're in the browser.

Reply:
This should just be grammar stuff. Does is change the meaning of the original text to you in some way? If it's only the comma versus period with the "that means" that's fine. Would be nice to break up the sentence a bit more if possible.

Reply:
I'm going to rewrite it